src/ngraph/runtime/dynamic/dynamic_backend.cpp

```diff
@@ -120,130 +120,207 @@ bool runtime::dynamic::DynamicExecutable::call(
     // (1) all shapes;
     // (2) all values of shape-relevant input tensors.
 
-    NGRAPH_CHECK(m_wrapped_function->get_parameters().size() == inputs.size());
-
-    std::vector<std::shared_ptr<runtime::Tensor>> wrapped_inputs;
-    std::vector<element::Type> arg_element_types;
-    std::vector<PartialShape> arg_shapes;
-
-    std::shared_ptr<Function> clone;
+    std::vector<int> merged_input_shapes;
+    std::ostringstream key;
+    size_t loop_count = 0;
+    for (auto& input : inputs)
     {
-        // We'll use AlignedBuffers to back the base pointers, storing them in this vector for RAII
-        // purposes.
-        std::vector<AlignedBuffer> arg_buffers;
-        arg_buffers.reserve(inputs.size());
-        std::vector<void*> arg_value_base_pointers(inputs.size());
-
-        size_t i = 0;
-
-        for (auto& input : inputs)
+        if (m_wrapped_function->get_parameters()[loop_count]->is_relevant_to_shapes())
         {
-            if (m_wrapped_function->get_parameters()[i]->is_relevant_to_shapes())
+            // Caching on values of Shape relevant inputs
+            void* data_ptr = malloc(input->get_size_in_bytes());
```

Comment:
Really, just use a vector<int64_t> here instead of malloc. malloc is old-school and dangerous. You could also use an AlignedBuffer which is more efficient that a buffer for this use and is RAII so you don't need to worry about deleting it.